--- a/luminol-server/build.gradle.kts
+++ b/luminol-server/build.gradle.kts
@@ -48,7 +_,23 @@
         }
     }
 
-    activeFork = luminol
+    val shiroha = forks.register("shiroha") {
+        forks = luminol
+        upstream.patchRepo("paperServer") {
+            upstreamRepo = luminol.patchedRepo("paperServer")
+            patchesDir = rootDirectory.dir("shiroha-server/paper-patches")
+            outputDir = rootDirectory.dir("paper-server")
+        }
+
+        upstream.patchDir("luminolServer") {
+            upstreamPath = "luminol-server"
+            excludes = setOf("src/minecraft", "paper-patches", "minecraft-patches", "build.gradle.kts", "build.gradle.kts.patch")
+            patchesDir = rootDirectory.dir("shiroha-server/luminol-patches")
+            outputDir = rootDirectory.dir("luminol-server")
+        }
+    }
+
+    activeFork = shiroha
 
 
     spigot {
@@ -139,12 +_,16 @@
         resources { srcDir("../paper-server/src/main/resources") }
         java { srcDir("../folia-server/src/main/java") }
         resources { srcDir("../folia-server/src/main/resources") }
+        java { srcDir("../luminol-server/src/main/java") }
+        resources { srcDir("../luminol-server/src/main/resources") }
     }
     test {
         java { srcDir("../paper-server/src/test/java") }
         resources { srcDir("../paper-server/src/test/resources") }
         java { srcDir("../folia-server/src/test/java") }
         resources { srcDir("../folia-server/src/test/resources") }
+        java { srcDir("../luminol-server/src/main/java") }
+        resources { srcDir("../luminol-server/src/main/resources") }
     }
 }
 val log4jPlugins = sourceSets.create("log4jPlugins") {
@@ -172,7 +_,10 @@
 }
 
 dependencies {
-    implementation(project(":luminol-api")) // Luminol
+    implementation(project(":shiroha-api")) // Luminol // Shiroha
+    // Shiroha start - Dependenices
+    implementation(project(":FlowSched"))
+    // Shiroha end
     // Luminol start - Dependenices insert
     implementation("io.netty:netty-all:4.2.9.Final") // used for io_uring
     implementation("com.electronwill.night-config:toml:3.8.3") // Night config
@@ -273,14 +_,14 @@
         val gitBranch = git.exec(providers, "rev-parse", "--abbrev-ref", "HEAD").get().trim()
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
-            "Implementation-Title" to "Luminol",
+            "Implementation-Title" to "Shiroha",
             "Implementation-Version" to implementationVersion,
             "Implementation-Vendor" to date,
-            "Specification-Title" to "Luminol",
+            "Specification-Title" to "Shiroha",
             "Specification-Version" to project.version,
-            "Specification-Vendor" to "LuminolMC org",
-            "Brand-Id" to "luminolmc:luminol",
-            "Brand-Name" to "Luminol",
+            "Specification-Vendor" to "StarCraft",
+            "Brand-Id" to "starcraft:shiroha",
+            "Brand-Name" to "Shiroha",
             "Build-Number" to (build ?: ""),
             "Build-Time" to buildTime.toString(),
             "Git-Branch" to gitBranch,
@@ -432,7 +_,7 @@
 }
 
 fill {
-    project("luminol")
+    project("shiroha")
     versionFamily(paperweight.minecraftVersion.map { it.split(".", "-").takeWhile { part -> part.toIntOrNull() != null }.take(2).joinToString(".") })
     version(paperweight.minecraftVersion)
 
