From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mrhua269 <mrhua269@gmail.com>
Date: Fri, 5 Dec 2025 20:28:38 +0800
Subject: [PATCH] Replace moonrise's worker with flowsched


diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/ChunkTaskScheduler.java b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/ChunkTaskScheduler.java
index 65716aa20fec95533249ba39bbf10dd570c7c64f..ed1d41ff088a7744eec5c5839813b584f84a9bfe 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/ChunkTaskScheduler.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/ChunkTaskScheduler.java
@@ -26,7 +26,11 @@ import ca.spottedleaf.moonrise.patches.chunk_system.ticket.ChunkSystemTicketType
 import ca.spottedleaf.moonrise.patches.chunk_system.util.ParallelSearchRadiusIteration;
 import com.google.gson.JsonArray;
 import com.google.gson.JsonObject;
+import com.ishland.c2me.base.common.scheduler.SchedulingManager;
+import com.ishland.c2me.base.common.scheduler.SingleThreadExecutor;
+import com.ishland.flowsched.executor.ExecutorManager;
 import com.mojang.logging.LogUtils;
+import io.papermc.paper.configuration.GlobalConfiguration;
 import net.minecraft.CrashReport;
 import net.minecraft.CrashReportCategory;
 import net.minecraft.ReportedException;
@@ -61,6 +65,10 @@ import java.util.Objects;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.function.Consumer;
+// Shiroha start - replace moonrise executors with flowsched
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicInteger;
+// Shiroha end
 
 public final class ChunkTaskScheduler {
 
@@ -115,6 +123,12 @@ public final class ChunkTaskScheduler {
     }
 
     public final ServerLevel world;
+    // Shiroha start - replace moonrise executors with flowsched
+    private final AtomicInteger prioritizedSchedulerCounter = new AtomicInteger(0);
+    private final Executor managerThread;
+    public final ExecutorManager executorManager;
+    public final SchedulingManager schedulingManager;
+    // Shiroha end
     public final RadiusAwarePrioritisedExecutor radiusAwareScheduler;
     public final PrioritisedThreadPool.ExecutorGroup.ThreadPoolExecutor parallelGenExecutor;
     private final PrioritisedThreadPool.ExecutorGroup.ThreadPoolExecutor radiusAwareGenExecutor;
@@ -301,6 +315,18 @@ public final class ChunkTaskScheduler {
         this.compressionExecutor = MoonriseCommon.LOAD_GROUP.createExecutor(-1, MoonriseCommon.WORKER_QUEUE_HOLD_TIME, 0);
         this.saveExecutor = MoonriseCommon.LOAD_GROUP.createExecutor(-1, MoonriseCommon.WORKER_QUEUE_HOLD_TIME, 0);
         this.chunkHolderManager = new ChunkHolderManager(world, this);
+        // Shiroha start - replace moonrise executors with flowsched
+        final SingleThreadExecutor manager = new SingleThreadExecutor();
+        manager.setDaemon(true);
+        manager.setName("c2me-sched");
+        manager.start();
+        this.managerThread = manager;
+        this.executorManager = new ExecutorManager(GlobalConfiguration.get().chunkSystem.workerThreads, thread -> {
+            thread.setDaemon(true);
+            thread.setName("c2me-worker-%d".formatted(prioritizedSchedulerCounter.getAndIncrement()));
+        });
+        this.schedulingManager = new SchedulingManager(this.managerThread, this.executorManager);
+        // Shiroha end
     }
 
     private final AtomicBoolean failedChunkSystem = new AtomicBoolean();
@@ -872,6 +898,7 @@ public final class ChunkTaskScheduler {
         this.radiusAwareGenExecutor.halt();
         this.parallelGenExecutor.halt();
         this.loadExecutor.halt();
+        this.executorManager.shutdown(); // Shiroha - replace moonrise executors with flowsched
         if (sync) {
             final long time = System.nanoTime();
             for (long failures = 9L;; failures = ConcurrentUtil.linearLongBackoff(failures, 500_000L, 50_000_000L)) {
diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/task/ChunkLoadTask.java b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/task/ChunkLoadTask.java
index 1440c9e2b106616884edcb20201113320817ed9f..2a09b55e641e4bebe071c7f8dfc047bc4c20e7fe 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/task/ChunkLoadTask.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/task/ChunkLoadTask.java
@@ -27,6 +27,9 @@ import java.lang.invoke.VarHandle;
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Consumer;
+// Shiroha start - replace moonrise executors with flowsched
+import i.moniasuki.shiroha.utils.FlowSchedRunnableTask2CCTask;
+// Shiroha end
 
 public final class ChunkLoadTask extends ChunkProgressionTask {
 
@@ -294,7 +297,10 @@ public final class ChunkLoadTask extends ChunkProgressionTask {
 
         @Override
         protected PrioritisedExecutor.PrioritisedTask createOffMain(final Runnable run, final Priority priority) {
-            return this.scheduler.loadExecutor.createTask(run, priority);
+            // Shiroha start - replace moonrise executors with flowsched
+            // return this.scheduler.loadExecutor.createTask(run, priority);
+            return new FlowSchedRunnableTask2CCTask(run, this.scheduler.schedulingManager, priority);
+            // Shiroha end
         }
 
         @Override
@@ -391,7 +397,10 @@ public final class ChunkLoadTask extends ChunkProgressionTask {
 
         @Override
         protected PrioritisedExecutor.PrioritisedTask createOffMain(final Runnable run, final Priority priority) {
-            return this.scheduler.loadExecutor.createTask(run, priority);
+            // Shiroha start - replace moonrise executors with flowsched
+            // return this.scheduler.loadExecutor.createTask(run, priority);
+            return new FlowSchedRunnableTask2CCTask(run, this.scheduler.schedulingManager, priority);
+            // Shiroha end
         }
 
         @Override
@@ -453,7 +462,10 @@ public final class ChunkLoadTask extends ChunkProgressionTask {
 
         @Override
         protected PrioritisedExecutor.PrioritisedTask createOffMain(final Runnable run, final Priority priority) {
-            return this.scheduler.loadExecutor.createTask(run, priority);
+            // Shiroha start - replace moonrise executors with flowsched
+            // return this.scheduler.loadExecutor.createTask(run, priority);
+            return new FlowSchedRunnableTask2CCTask(run, this.scheduler.schedulingManager, priority);
+            // Shiroha end
         }
 
         @Override
diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/task/ChunkUpgradeGenericStatusTask.java b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/task/ChunkUpgradeGenericStatusTask.java
index 25d8da4773dcee5096053e7e3788bfc224d705a7..37bb7369ef243b18b12f4aba37d9e8ab83845a24 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/task/ChunkUpgradeGenericStatusTask.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/task/ChunkUpgradeGenericStatusTask.java
@@ -23,6 +23,11 @@ import java.lang.invoke.VarHandle;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+// Shiroha start - replace moonrise executors with flowsched
+import net.minecraft.world.level.ChunkPos;
+import i.moniasuki.shiroha.utils.FlowSchedRunnableTask2CCTask;
+import i.moniasuki.shiroha.utils.FlowSchedScheduledTask2CCTask;
+// Shiroha end
 
 public final class ChunkUpgradeGenericStatusTask extends ChunkProgressionTask implements Runnable {
 
@@ -47,13 +52,22 @@ public final class ChunkUpgradeGenericStatusTask extends ChunkProgressionTask im
         this.toStatus = toStatus;
         this.neighbours = neighbours;
         if (((ChunkSystemChunkStatus)this.toStatus).moonrise$isParallelCapable()) {
-            this.generateTask = this.scheduler.parallelGenExecutor.createTask(this, priority);
+            // Shiroha start - replace moonrise executors with flowsched
+            // this.generateTask = this.scheduler.parallelGenExecutor.createTask(this, priority);
+            this.generateTask = new FlowSchedRunnableTask2CCTask(this, scheduler.schedulingManager);
+            // Shiroha end
         } else {
             final int writeRadius = ((ChunkSystemChunkStatus)this.toStatus).moonrise$getWriteRadius();
             if (writeRadius < 0) {
-                this.generateTask = this.scheduler.radiusAwareScheduler.createInfiniteRadiusTask(this, priority);
+                // Shiroha start - replace moonrise executors with flowsched
+                // this.generateTask = this.scheduler.radiusAwareScheduler.createInfiniteRadiusTask(this, priority);
+                throw new IllegalArgumentException("???");
+                // Shiroha end
             } else {
-                this.generateTask = this.scheduler.radiusAwareScheduler.createTask(chunkX, chunkZ, writeRadius, this, priority);
+                // Shiroha start - replace moonrise executors with flowsched
+                // this.generateTask = this.scheduler.radiusAwareScheduler.createTask(chunkX, chunkZ, writeRadius, this, priority);
+                this.generateTask = new FlowSchedScheduledTask2CCTask(this, new ChunkPos(chunkX, chunkZ), writeRadius, scheduler.schedulingManager);
+                // Shiroha end
             }
         }
     }
diff --git a/ca/spottedleaf/moonrise/patches/starlight/light/StarLightInterface.java b/ca/spottedleaf/moonrise/patches/starlight/light/StarLightInterface.java
index 51f4dd4f583dfbd16cb00f1cb4418d1044cecb1c..f21dc5f917e24b5839dd0f434f5adeeb24ec891b 100644
--- a/ca/spottedleaf/moonrise/patches/starlight/light/StarLightInterface.java
+++ b/ca/spottedleaf/moonrise/patches/starlight/light/StarLightInterface.java
@@ -37,6 +37,10 @@ import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
 import java.util.function.IntConsumer;
+// Shiroha start - replace moonrise executors with flowsched
+import i.moniasuki.shiroha.utils.FlowSchedRunnableTask2CCTask;
+import i.moniasuki.shiroha.utils.FlowSchedScheduledTask2CCTask;
+//Shiroha end
 
 public final class StarLightInterface {
 
@@ -885,10 +889,16 @@ public final class StarLightInterface {
             public ServerChunkTasks(final long chunkCoordinate, final StarLightInterface lightEngine,
                                     final ServerLightQueue queue, final Priority priority) {
                 super(chunkCoordinate, lightEngine, queue);
-                this.task = ((ChunkSystemServerLevel)(ServerLevel)lightEngine.getWorld()).moonrise$getChunkTaskScheduler().radiusAwareScheduler.createTask(
+                // Shiroha start - replace moonrise executors with flowsched
+                /*this.task = ((ChunkSystemServerLevel)(ServerLevel)lightEngine.getWorld()).moonrise$getChunkTaskScheduler().radiusAwareScheduler.createTask(
                         CoordinateUtils.getChunkX(chunkCoordinate), CoordinateUtils.getChunkZ(chunkCoordinate),
                         ((ChunkSystemChunkStatus)ChunkStatus.LIGHT).moonrise$getWriteRadius(), this, priority
-                );
+                );*/
+                this.task = new FlowSchedScheduledTask2CCTask(this,
+                        new ChunkPos(CoordinateUtils.getChunkX(chunkCoordinate), CoordinateUtils.getChunkZ(chunkCoordinate)),
+                        ((ChunkSystemChunkStatus)ChunkStatus.LIGHT).moonrise$getWriteRadius(),
+                        ((ChunkSystemServerLevel)(ServerLevel)lightEngine.getWorld()).moonrise$getChunkTaskScheduler().schedulingManager);
+                // Shiroha end
             }
 
             public boolean markTicketAdded() {
diff --git a/io/papermc/paper/FeatureHooks.java b/io/papermc/paper/FeatureHooks.java
index df6fbb35e5023b42de0b97434712e04a6b3e66a3..b7ee6a732aae55a8aff997a1c94276ddd1efb027 100644
--- a/io/papermc/paper/FeatureHooks.java
+++ b/io/papermc/paper/FeatureHooks.java
@@ -47,7 +47,7 @@ public final class FeatureHooks {
     }
 
     public static void registerPaperCommands(final Map<Set<String>, PaperSubcommand> commands) {
-        commands.put(Set.of("fixlight"), new FixLightCommand()); // Paper - rewrite chunk system
+        // commands.put(Set.of("fixlight"), new FixLightCommand()); // Paper - rewrite chunk system // Shiroha - Replace moonrise executors with flow sched
         commands.put(Set.of("debug", "chunkinfo", "holderinfo"), new ChunkDebugCommand());  // Paper - rewrite chunk system
     }
 
